File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/LedgerClient.java
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.rxjava;

/** Contains the set of services provided by a Ledger implementation */
public interface LedgerClient {

  /** @return The identifier of the Ledger connected to this {@link LedgerClient} */
  StateClient getStateClient();

  UpdateClient getTransactionsClient();

  CommandClient getCommandClient();

  CommandCompletionClient getCommandCompletionClient();

  CommandSubmissionClient getCommandSubmissionClient();

  EventQueryClient getEventQueryClient();

  PackageClient getPackageClient();

  TimeClient getTimeClient();

  UserManagementClient getUserManagementClient();
}

File: ledger-service/http-json-ledger-client/ce/src/main/scala/com/daml/http/LedgerClient.scala
// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.http

import com.daml.ledger.client.configuration.LedgerClientChannelConfiguration
import io.grpc.netty.NettyChannelBuilder

import scala.concurrent.{ExecutionContext, Future}

object LedgerClient extends LedgerClientBase {

  def channelBuilder(
      ledgerHost: String,
      ledgerPort: Int,
      clientChannelConfig: LedgerClientChannelConfiguration,
  )(implicit executionContext: ExecutionContext): Future[NettyChannelBuilder] =
    Future(clientChannelConfig.builderFor(ledgerHost, ledgerPort))

}
